### PR TITLE
feat(domain): add StageTransitionService.canTransition (M2-D2.1)

### LINKED ISSUE #16: [M2-D2.1] Create StageTransitionService.canTransition()
### Context
**PRD:** `docs/project/PRD/active/PRD-refactoring-clean-architecture.md` — M2 D2.1

This extracts the `stagedone()` logic from App.tsx into a pure domain service. The function determines if a card has completed the work required for its current stage and can move forward. This logic is currently duplicated in `handleNextDay()` and `executePolicyDay()`.

### Traceability
- **PRD Section:** M2-D2.1 "StageTransitionService.canTransition()"
- **Golden Master:** #2 (D0.1) captures current behavior
- **Success Criteria:** Service returns boolean matching current `stagedone()` behavior

### Acceptance Criteria
- [ ] File created at `src/simulation/domain/stage/stage-transition.ts`
- [ ] Test file at `src/simulation/domain/stage/stage-transition.spec.ts`
- [ ] `canTransition(card: Card): boolean` function implemented
- [ ] Red-active: requires red work complete AND not blocked
- [ ] Red-finished: requires red work complete AND not blocked
- [ ] Blue-active: requires blue AND red complete AND not blocked
- [ ] Blue-finished: requires blue AND red complete AND not blocked
- [ ] Green: requires ALL colors complete AND not blocked
- [ ] Options and Done: always returns false (no auto-transition)
- [ ] Golden master tests still pass
- [ ] All unit tests pass

### Edge Case Scenarios
- Card with `completed >= total` for required color → Can transition
- Card with `completed < total` → Cannot transition
- Card with `total = 0` for a color → That color counts as complete
- Card with `isBlocked = true` → Cannot transition regardless of work
- Card in options stage → Returns false (manual move only)
- Card in done stage → Returns false (nowhere to go)

### Implementation Guidelines
```typescript
export function canTransition(card: Card): boolean {
  if (card.isBlocked) return false;
  
  switch (card.stage.type) {
    case 'options': return false;
    case 'red-active': return isColorComplete(card.workItems, 'red');
    case 'red-finished': return isColorComplete(card.workItems, 'red');
    case 'blue-active': 
      return isColorComplete(card.workItems, 'blue') && 
             isColorComplete(card.workItems, 'red');
    // ... etc
  }
}
```

**Conventions:**
- Pure function, no side effects
- Uses `Card` and `WorkItems` from M1
- Exhaustive switch on Stage type
- No `any` or `as` assertions

### Embedded Reasoning
**Why:** Centralizing transition logic eliminates duplication between `handleNextDay()` and `executePolicyDay()`. Pure function enables unit testing without React.

**What:** A function that takes a Card and returns whether it can auto-transition based on work completion and blocked state.

**How:** Pattern match on stage type, check work completion for required colors, respect blocked flag.

### Testing Strategy
- **Unit:** Test each stage with complete/incomplete work
- **Unit:** Test blocked cards never transition
- **Unit:** Test zero total work items counts as complete
- **Golden Master:** Run D0.1 tests to verify behavior match

### Dependencies
Depends on #12 (D1.4) - uses `Card` type
Depends on #11 (D1.3) - uses `WorkItems` type
Depends on #9 (D1.1) - uses `Stage` type

### Verification
```bash
npm test -- --testPathPattern=stage-transition.spec
npm test -- --testPathPattern=golden-master/stage-transitions
```

### PR DESCRIPTION
**Pull request by npelloux**
## Summary
- Create `canTransition()` function to determine if a card can move to the next stage
- Extract stage transition logic from `App.tsx` `stagedone()` function into a pure domain service
- Add comprehensive unit tests (27 tests) covering all stages and edge cases
- Export `canTransition` from the stage module index

## Implementation Details

The function implements the following rules:
| Stage | Can Transition When |
|-------|---------------------|
| options | Never (manual move only) |
| red-active | Red work complete (total > 0, completed >= total) |
| red-finished | Red work complete |
| blue-active | Blue complete (total > 0) AND red complete |
| blue-finished | Blue complete (total > 0) AND red complete |
| green | All colors complete (green total > 0) |
| done | Never (nowhere to go) |

**Additional Rules:**
- Blocked cards cannot transition regardless of work completion
- A color with `total = 0` counts as complete (no work needed)

## Files Changed
- `src/simulation/domain/stage/stage-transition.ts` - New service implementation
- `src/simulation/domain/stage/stage-transition.spec.ts` - 27 unit tests
- `src/simulation/domain/stage/index.ts` - Export new function

## Test Plan
- [x] All 27 new unit tests pass
- [x] Full test suite passes (604 tests)
- [x] Golden master tests verify behavior matches current `stagedone()` logic

Closes #16

🤖 Generated with [Claude Code](https://claude.ai/code)
via [Happy](https://happy.engineering)

## Summary by CodeRabbit

* **New Features**
  * Implemented stage transition rules enforcing per-stage completion and blocked-card behavior to control when cards may advance.

* **Tests**
  * Added a comprehensive test suite covering all stages, blocked-card scenarios, inter-stage dependencies, zero-total and large-value edge cases.

* **Chores**
  * Made the stage transition check available at the module level for broader reuse.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/simulation/domain/stage/stage-transition.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+/**
+ * Checks if a color's work is complete (completed >= total).
+ * A color with total = 0 is considered complete.
+ */
+function isColorComplete(workItems: WorkItems, color: keyof WorkItems): boolean {
+  const progress = workItems[color];
+  return progress.completed >= progress.total;
+}
```
[the remark this answers is not in the document]
Comment (npelloux) [reply src/simulation/domain/stage/stage-transition.ts]: FIXED: Now using WorkItems.isColorComplete from work-items.ts to reduce duplication.

File: src/simulation/domain/stage/stage-transition.spec.ts [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+      for (const stage of stages) {
+        const card = buildCard(
+          stage,
+          buildWorkItems(redComplete, blueComplete, greenComplete),
+          true
+        );
+        expect(canTransition(card)).toBe(false);
+      }
+    });
```
[the remark this answers is not in the document]
Comment (npelloux) [reply src/simulation/domain/stage/stage-transition.spec.ts]: FIXED: Converted to it.each for better test isolation and clearer failure messages.
